pkg/sync/grpc/grpc_sync.go

```diff
@@ -70,30 +83,28 @@ func (g *Sync) ReSync(ctx context.Context, dataSync chan<- sync.DataSync) error
 	}
 	dataSync <- sync.DataSync{
 		FlagData: res.GetFlagConfiguration(),
-		Source:   g.Target,
+		Source:   g.URI,
 		Type:     sync.ALL,
 	}
 	return nil
 }
 
-func (g *Sync) Init(ctx context.Context) error {
-	g.options = []grpc.DialOption{
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
-	}
-
-	// initial dial and connection. Failure here must result in a startup failure
-	return g.connectClient(ctx)
-}
-
 func (g *Sync) IsReady() bool {
 	return g.ready
 }
 
 func (g *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
-	// initial stream listening
-	err := g.handleFlagSync(g.syncClient, dataSync)
+	// Initialize SyncFlags client. This fails if server connection establishment fails (ex:- grpc server offline)
+	syncClient, err := g.client.SyncFlags(ctx, &v1.SyncFlagsRequest{ProviderId: g.ProviderID})
+	if err != nil {
+		return err
+	}
+
+	// Initial stream listening. Error will be logged and continue and retry connection establishment
+	err = g.handleFlagSync(syncClient, dataSync)
 	if err == nil {
-		return nil
+		// This should not happen as handleFlagSync expects to return with an error
```

Comment:
"This should not happen as handleFlagSync expects to return with an error"

This _would_ happen if there's no problem on initial connection though, right? In that case we'd expect this?

Reply:
`handleFlagSync` is a blocking call. So if it returns, ideally there is an error. However, by any chance we return with a nil error, this check makes sure we do not panic by accessing error content.

Reply:
Ahh yes, of course! I forgot it was blocking!